src/OrchardCore.Modules/OrchardCore.Media/Indexing/IMediaFileTextProvider.cs

```diff
@@ -0,0 +1,22 @@
+using System.IO;
+
+namespace OrchardCore.Media.Indexing
+{
+    public interface IMediaFileTextProvider
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// <para>
+        /// A separate method is necessary since an empty or null return value from <see cref="GetText(string, Stream)"/>
+        /// can simply mean the file's textual content is empty.
+        /// </para>
+        /// </remarks>
+        bool CanHandle(string path);
```

Comment:
This could be a utility or `private` method that's utilized by `GetText()`, but I will leave others to comment

Reply:
Seems it's used to filter `IMediaFileTextProvider`s, right?

Reply:
Yes, and ;):

![image](https://user-images.githubusercontent.com/1976647/141661411-c0180a80-cc10-41ee-b8d5-b71a6649f7e3.png)